File: 435. Non-overlapping Intervals.cpp
class Solution {
public:
    int eraseOverlapIntervals(std::vector<std::vector<int>>& intervals) {
        std::sort(intervals.begin(), intervals.end(), [](const std::vector<int>& a, const std::vector<int>& b) {
            return a[1] < b[1];
        });

        int n = intervals.size();
        int ans = 0;
        int last = 0;

        for (int i = 1; i < n; ++i) {
            if (intervals[i][0] < intervals[last][1]) {
                ++ans;
            } else {
                last = i;
            }
        }

        return ans;
    }
};

File: 435. Non-overlapping Intervals.java
class Solution {
    public int eraseOverlapIntervals(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> Integer.compare(a[0], b[0]));
        int count = 0;
        int x1 = intervals[0][0];
        int y1 = intervals[0][1];
        // System.out.println(y1);
        for(int i = 1; i < intervals.length; i++){
            int x2 = intervals[i][0];
            int y2 = intervals[i][1];
            if(y1 <= x2){
                x1 = x2;
                y1 = y2;
            }
            else if(y1 > y2){
                y1 = y2;
                count++;
            }
            else count++;
        }

        return count;
    }
}